src/rgw/rgw_acl.h

```diff
@@ -60,10 +60,8 @@ class ACLGrant
     }
   }
 
-  ACLGranteeType& get_type() { return type; }
-  const ACLGranteeType& get_type() const { return type; }
-  ACLPermission& get_permission() { return permission; }
-  const ACLPermission& get_permission() const { return permission; }
+  ACLGranteeType get_type() const { return type; }
```

Comment:
how big are these types?
why not return a ref?

Reply:
ACLPermission and ACLGranteeType are int-sized wrapper classes defined in rgw_acl_types.h. since they fit in a register, value semantics are more efficient than references. nothing relied on the mutable versions, ie `grant.get_permission() = 5`

Reply:
but there are places in the code where they are passed as const ref to a function. these should probably also be changed to be passed by-value

Reply:
ok, i'll hunt those down